### PR TITLE
Added Go SDK examples

### PR DESCRIPTION
**Pull request by jasonjoh**

### CODE REVIEW

File: concepts/sdks/customize-client.md [hunk 1]
```diff
@@ -117,6 +117,46 @@ final GraphServiceClient graphServiceClient = GraphServiceClient
...
+## [Go](#tab/Go)
+
+[!INCLUDE [go-sdk-preview](../../includes/go-sdk-preview.md)]
+
+```go
+// Auth provider
+auth, err := a.NewAzureIdentityAuthenticationProviderWithScopes(...)
+
+clientOptions := core.GraphClientOptions{
+    GraphServiceVersion:        "",
```
Comment (baywet) [inline concepts/sdks/customize-client.md:129]: could use msgraphsdk.clientOptions if I changed the c for a C instead, what do you think?

File: concepts/sdks/customize-client.md [hunk 2]
```diff
@@ -117,6 +117,46 @@ final GraphServiceClient graphServiceClient = GraphServiceClient
...
+
+// Get default middleware from SDK
+defaultMiddleware := core.GetDefaultMiddlewaresWithOptions(&clientOptions)
+
+// Get instance of custom middleware
+// Implement a custom middleware by implementing the Middleware interface
+// https://github.com/microsoft/kiota/blob/main/http/go/nethttp/middleware.go
+customMiddleware := []khttp.Middleware{mycustom.NewCustomHandler()}
+
+allMiddleware := make([]khttp.Middleware, len(defaultMiddleware)+1)
```
Comment (baywet) [inline concepts/sdks/customize-client.md:141]: I believe you should be able to get around by doing

```golang
allMiddleware := []khttp.Middleware {
   mycustom.NewCustomHandler(),
   defaultMiddleware...
}
```

And drop the copy etc... (I haven't tested the code though)

**Review by baywet — CHANGES_REQUESTED** [review]
thanks for taking the time to put all of that together! a few minor comments. Let me make the changes in the service libs so you can update the version thing

File: concepts/sdks/customize-client.md (hunk 1 above)
Comment (baywet) [reply concepts/sdks/customize-client.md:129]: https://github.com/microsoftgraph/msgraph-beta-sdk-go/pull/15
https://github.com/microsoftgraph/msgraph-sdk-go/pull/22

Comment (jasonjoh) [reply concepts/sdks/customize-client.md:129]: Sure if you think that's the way to go I'm all for it.

File: concepts/sdks/customize-client.md (hunk 2 above)
Comment (jasonjoh) [reply concepts/sdks/customize-client.md:141]: Ah yeah, thanks for that.

Comment (jasonjoh) [reply concepts/sdks/customize-client.md]: Should be resolved now

File: concepts/sdks/customize-client.md (hunk 1 above)
Comment (jasonjoh) [reply concepts/sdks/customize-client.md]: Should be resolved now

File: concepts/sdks/customize-client.md [hunk 3]
```diff
@@ -117,6 +117,44 @@ final GraphServiceClient graphServiceClient = GraphServiceClient
...
+    khttp "github.com/microsoft/kiota/http/go/nethttp"
+    msgraphsdk "github.com/microsoftgraph/msgraph-sdk-go"
+    core "github.com/microsoftgraph/msgraph-sdk-go-core"
+)
+
+// Auth provider
+auth, err := a.NewAzureIdentityAuthenticationProviderWithScopes(...)
+
+// Get default middleware from SDK
+defaultMiddleware := core.GetDefaultMiddlewaresWithOptions(core.ClientOptions)
```
Comment (baywet) [inline concepts/sdks/customize-client.md:136]:
```suggestion
defaultMiddleware := core.GetDefaultMiddlewaresWithOptions(msgraphsdk.ClientOptions)
```

File: concepts/sdks/customize-client.md [hunk 4]
```diff
@@ -117,6 +117,44 @@ final GraphServiceClient graphServiceClient = GraphServiceClient
...
+
+// Get instance of custom middleware
+// Implement a custom middleware by implementing the Middleware interface
+// https://github.com/microsoft/kiota/blob/main/http/go/nethttp/middleware.go
+allMiddleware := []khttp.Middleware{
```
Comment (baywet) [inline concepts/sdks/customize-client.md:141]:
```suggestion
allMiddleware := append(defaultMiddleware, mycustom.NewCustomHandler())
```

File: concepts/sdks/customize-client.md [hunk 5]
```diff
@@ -117,6 +117,44 @@ final GraphServiceClient graphServiceClient = GraphServiceClient
...
+    mycustom.NewCustomHandler(),
```
Comment (baywet) [inline concepts/sdks/customize-client.md:142]:
```suggestion
```

File: concepts/sdks/customize-client.md [hunk 6]
```diff
@@ -117,6 +117,44 @@ final GraphServiceClient graphServiceClient = GraphServiceClient
...
+    defaultMiddleware...
```
Comment (baywet) [inline concepts/sdks/customize-client.md:143]:
```suggestion
```

File: concepts/sdks/customize-client.md [hunk 7]
```diff
@@ -117,6 +117,44 @@ final GraphServiceClient graphServiceClient = GraphServiceClient
...
+}
```
Comment (baywet) [inline concepts/sdks/customize-client.md:144]:
```suggestion
```

**Review by baywet — CHANGES_REQUESTED** [review]

**Review by baywet — APPROVED** [review]

File: concepts/sdks/customize-client.md [hunk 8]
```diff
@@ -117,6 +117,41 @@ final GraphServiceClient graphServiceClient = GraphServiceClient
...
+    khttp "github.com/microsoft/kiota/http/go/nethttp"
+    msgraphsdk "github.com/microsoftgraph/msgraph-sdk-go"
+    core "github.com/microsoftgraph/msgraph-sdk-go-core"
+)
+
+// Auth provider
+auth, err := a.NewAzureIdentityAuthenticationProviderWithScopes(...)
+
+// Get default middleware from SDK
+defaultMiddleware := core.GetDefaultMiddlewaresWithOptions(msgraphsdk.ClientOptions)
```
Comment (baywet) [inline concepts/sdks/customize-client.md:136]:
```suggestion
defaultMiddleware := core.GetDefaultMiddlewaresWithOptions(msgraphsdk.GetDefaultClientOptions())
```
```suggestion
defaultMiddleware := core.GetDefaultMiddlewaresWithOptions(msgraphsdk.ClientOptions)
```

**Pull request merged**